src/batches/setBatchImageCalculation.m

```diff
@@ -16,22 +16,70 @@
   % :type outDir: string
   % :param expression: mathematical expression to apply (for example '(i1+i2)>3')
   % :type expression: string
+  % :param expression: data type that must be one of the following:
+  %    - 'uint8'
+  %    - 'int16' (default)
+  %    - 'int32'
+  %    - 'float32'
+  %    - 'float64'
+  %    - 'int8'
+  %    - 'uint16'
+  %    - 'uint32'
+  % :type expression: string
+  %
+  % See ``spm_cfg_imcalc.m`` for more information::
+  %
+  %   ``edit(fullfile(spm('dir'), 'config', 'spm_cfg_imcalc.m'))``
+  %
+  %
   %
   % :returns: - :matlabbatch:
   %
   % (C) Copyright 2020 CPP_SPM developers
 
   printBatchName('image calculation');
 
-  matlabbatch{end + 1}.spm.util.imcalc.input = input;
-  matlabbatch{end}.spm.util.imcalc.output = output;
-  matlabbatch{end}.spm.util.imcalc.outdir = { outDir };
-  matlabbatch{end}.spm.util.imcalc.expression = expression;
+  allowedDataType = {'uint8', ...
+                     'int16', ...
+                     'int32', ...
+                     'float32', ...
+                     'float64', ...
+                     'int8', ...
+                     'uint16', ...
+                     'uint32'};
+
+  defaultDataType = 'int16';
```

Comment:
@CerenB you had set the value to `16` which would correspond to `'float32'`.

Should I change the default here?

Reply:
ah I see now! I do not know how/where would you like to call `defaultDataType` ?  Would we call it somewhere that value like `checkOptions.m` ?

For me, I need to set it to float32. Not sure for other users how it should be set.

Reply:
that's fine to just have the default in that file the way you changed it

if we run into trouble with that later, we have a way to access it via the input argument so we should be fine.